site/content/in-dev/unreleased/configuring-polaris-for-production.md

```diff
@@ -135,33 +135,19 @@ default, Polaris uses an in-memory metastore.
 > The default in-memory metastore is not suitable for production use, as it will lose all data
 > when the server is restarted; it is also unusable when multiple Polaris replicas are used.
 
-To use a durable metastore, you need to switch to the EclipseLink metastore, and provide your own
-`persistence.xml` file. This file contains details of the database used for metastore management and
-the connection settings. For more information, refer to the [metastore documentation]({{% ref
-"metastores" %}}).
+To use a durable metastore, you need to switch to the Relational JDBC backed metastore 
 
-Then, configure Polaris to use your metastore by setting the following properties:
+Configure the `polaris.persistence` section in your Polaris configuration file
+(`application.properties`) as follows:
```

Comment:
From my POV `application.properties` is not the best place for user-level settings. In binary distributions the default one is embedded into jar files and not directly accessible, while the optional local file is relative to the current directory, which is prone to runtime mistakes.

I'd suggest using java `-D` options or env. variables in user-facing docs.

Cf. https://quarkus.io/guides/config-reference

Reply:
I agree, as the whole page follows the theme of setting stuff in application.prop hence i kept this, please let me know what you think considering above !

Reply:
I only read the diff, not the whole doc :sweat_smile: but I keep my point. We do not have to fix the whole doc in this PR, though.

Reply:
No worries , how about i add another line below that in order to use the binaries configure using -D or env props ?

Reply:
Yeah this doc predates quarkus. Tbh I think it's kind of problematic that we don't offer a config file where you can change things without needing to rebuild. We should fix even if we first update our docs to reflect the current (bad) state

Reply:
In k8s it is pretty easy and natural to supply config as env. variables. However, if people like having the ability to use a local file for configuration, I guess it should not be too hard to support it with a predictable location (e.g. relative to the jar file or user's HOME dir as opposed to current dir). WDYT?

Reply:
In another PR, I proposed to introduce a "user facing" configuration (it's very easy to do that with Quarkus) and it's always possible to override via env variables or config maps on k8s.

Short term in this PR, I would recommend to document the env variable use, and if you think it makes sense I can revive my PR.

Reply:
documenting ENV vars makes sense to me, let me try doing that. 

Re: if people like having the ability to use a local file for configuration

IMHO, quarkus is quite configurable and its the deployement time they would need, loading confs from file will introduce these complexity like fallbacks of loading this file from [code](https://github.com/apache/polaris/blob/main/extension/persistence/eclipselink/src/main/java/org/apache/polaris/extension/persistence/impl/eclipselink/PolarisEclipseLinkPersistenceUnit.java#L129) never the less open the config override options for ex something configured in the file should override what the env specifies, by the time quarkus would get the handle of the file and apply configs, quarkus would have applied the confs to make DS ?

Reply:
Here is how it looks : 
![Screenshot 2025-05-05 at 10 56 42 AM](https://github.com/user-attachments/assets/0695629d-a134-4167-ad20-042ad9a59413)

Reply:
+1 to reviving @jbonofre 's PR : https://github.com/apache/polaris/pull/1032/files#diff-7f7a4d7e028cff5774c249e58114c43abf6c983ad8be3fc955abec59865272ba